components/src/lib/common/useModalState.ts

```diff
@@ -12,17 +12,14 @@ const useModalState = (
 ) => {
   const [isOpen, setIsOpen] = useState(initialIsOpen);
 
-  useEffect(() => {
-    onOpen();
-  }, []);
-
-  useEffect(() => {
-    if (isOpen) onOpen();
-    else onClose();
-  }, [isOpen]);
-
-  const openModal = () => setIsOpen(true);
-  const closeModal = () => setIsOpen(false);
+  const openModal = () => {
+    setIsOpen(true);
+    onOpen && onOpen();
```

Comment:
```suggestion
    onOpen?.();
```
[optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)을 사용하는 이런 방법도 있답니다!